### ISSUE TITLE
Spread operator is not transpiled down

### ISSUE THREAD

**Issue by flunderpero** [evanw/esbuild#333]
First off, great work you are doing with esbuild!

We still have to support Edge 18 and we are running into a problem with our ServiceWorker - we cannot register it. This is similar to this [SO thread](https://stackoverflow.com/questions/53628191/edge-script1028-expected-identifier-string-or-number). We used webpack before and hat to lower the target to `ES2017`.

I tried everything, but esbuild does not replace (all) spread operators in our build. I tried `ES2015`, `edge18` and almost everything in between. :-) The generated code still contains lots of spread operators.

This is my build-config:
```
    esbuild.buildSync({
        entryPoints: ["ervice_worker.ts"],
        format: "iife",
        tsconfig: "tsconfig.json",
        define: {
            window: "self",
            global: "self",
            // some more project-specific stuff ...
        },
        outdir: "dist",
        sourcemap: "external",
        minifyIdentifiers: false,
        minifySyntax: true,
        minifyWhitespace: true,
        bundle: true,
        external: ["http", "https", "crypto", "stream"],  // we use shared code with NodeJS and remove the imports later on
        target: "es2015",
    })
```

The resulting code contains array spreads (`[...a]`), function call spreads (`function a(...args)`) but afaics no object spreads (`{...o}`). 
Please note: Edge 18 does indeed support array and function call spreads, but *service workers* on Edge 18 do not.

**Comment by bathos** [comment]
> The resulting code contains array spreads ([...a]), function call spreads (function a(...args)) but afaics no object spreads ({...o}).

Iterable spread/rest was introduced in ES2015 and object spread/rest was introduced in ES2018. If the target is ES2015, that result sounds correct. ES5 was the last version of JS that didn't have any spread/rest.

**Comment by flunderpero** [comment]
Thanks for your reply. I didn't want to say that this is a bug per se. I will dig a bit deeper to find out what webpack with ts-loader and target 2017 does differently. The regular webapp-code is working fine, just the SW does not. We fought the same bug with webpack before.
(Edge <= and (especially) Safari are becoming the new IE5.5 :-) )

**Comment by flunderpero** [comment]
Ok, I found the problem. This time it had nothing to do with spread operators, but with a missing global reference. We have some code that is shared between Node and the web that uses `Buffer`.  Webpack automagically adds a polyfill for that (https://webpack.js.org/configuration/node/). I was aware of the fact, that esbuild does not do this and added the polyfill myself by importing it. This worked fine for the SPA build and for the SW build in all browsers but Edge 18. I think the import of the polyfill came after the global Buffer had already been referenced which lead to an error that Edge 18 completely eats up. That's why I jumped to the conclusion that we had the same problem with (object literal) spread operators as we previously had with Webpack. I am sorry for that.

I now added an `import {Buffer} from "buffer"` to every module that uses `Buffer` and it works.

Have you already thought about an option to add arbitrary JS to the top of a bundle? This would have been where I had put the import of the polyfill:
```
import {Buffer} from "buffer"
self.Buffer = Buffer  // ... the package "buffer" does not add a global reference to Buffer
```

I close this issue.
